instrumentation/src/javatests/org/oppia/android/instrumentation/player/ExplorationPlayerTest.kt

```diff
@@ -0,0 +1,154 @@
+package org.oppia.android.instrumentation.player
+
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import com.google.common.truth.Truth.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.findObjectByDesc
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.findObjectByRes
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.findObjectByText
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.scrollRecyclerViewTextIntoView
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.startOppiaFromScratch
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.waitForRes
+
+/** Tests for Explorations. */
+class ExplorationPlayerTest {
+  private lateinit var device: UiDevice
+
+  @Before
+  fun setUp() {
+    // Initialize UiDevice instance
+    device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+    device.startOppiaFromScratch()
+  }
+
+  @Test
+  fun testExploration_prototypeExploration_toolbarTitle_isDisplayedSuccessfully() {
+    navigateToPrototypeExploration()
+    assertThat(device.findObjectByRes("exploration_toolbar_title")).isNotNull()
+  }
+
+  @Test
+  fun testProtoTypeExploration_answerAllInteractionsAndCompleteExploration() {
+    navigateToPrototypeExploration()
+    completePrototypeExploration()
+
+    // Assert Topic Completed.
+    scrollRecyclerViewTextIntoView("Chapter 1: Prototype Exploration")
+    val chapterCompletedTick = device.findObjectByText(
+      "Chapter 1: Prototype Exploration"
+    )?.parent?.children?.get(2)
```

Comment:
Why 2? Seems like a magic number.

Reply:
![image](https://user-images.githubusercontent.com/54740946/129662757-7f7eda23-07b9-4247-98a5-4fdd125f62fc.png)
the `chapter_completed_tick` might be multiple if there are more than one chapters.

so I took the chapter title which is index 0 in the view group, took the parent reference to the the title, and then get its child at index 2 which makes it accurate

Reply:
Is there any way that we can retrieve the checkbox directly? It ought to at least have a content description for screenreader users if it doesn't already (/cc @rt4914). Could we add a content description & retrieve the object based on that?

I'm not keen on the '2' because it's non-specific (without this screenshot, someone coming across this isn't going to have any idea why or if this is correct), and it's a bit fragile (if we move elements in the recyclerview around, that almost guarantees breaking this test even if the UX is still more or less the same).

Reply:
I agree, actually the content description is a bit off here, it shows %s instead of actual values.
Maybe add a todo to keep track of it of fix it in this pr

Reply:
Done, I had some reference for this

Reply:
![image](https://user-images.githubusercontent.com/54740946/130023174-ca1a5513-024b-43c0-bdb3-9bc47446607f.png)

Reply:
Ah, thanks for fixing that! Can we retrieve the object now by its content description rather than using the magic number?

Also strangely, the full deltas for this PR are not showing these changes from your latest commit. Any idea why?